pkg/apis/pipeline/v1beta1/resource_types.go

```diff
@@ -117,57 +115,6 @@ type PipelineResourceBinding struct {
 	ResourceSpec *resource.PipelineResourceSpec `json:"resourceSpec,omitempty"`
 }
 
-// PipelineResourceResult used to export the image name and digest as json
-type PipelineResourceResult struct {
-	Key          string     `json:"key"`
-	Value        string     `json:"value"`
-	ResourceName string     `json:"resourceName,omitempty"`
-	ResultType   ResultType `json:"type,omitempty"`
-}
-
-// ResultType used to find out whether a PipelineResourceResult is from a task result or not
-// Note that ResultsType is another type which is used to define the data type
-// (e.g. string, array, etc) we used for Results
-type ResultType int
-
-// UnmarshalJSON unmarshals either an int or a string into a ResultType. String
-// ResultTypes were removed because they made JSON messages bigger, which in
-// turn limited the amount of space in termination messages for task results. String
-// support is maintained for backwards compatibility - the Pipelines controller could
-// be stopped midway through TaskRun execution, updated with support for int in place
-// of string, and then fail the running TaskRun because it doesn't know how to interpret
-// the string value that the TaskRun's entrypoint will emit when it completes.
-func (r *ResultType) UnmarshalJSON(data []byte) error {
-	var asInt int
-	var intErr error
-
-	if err := json.Unmarshal(data, &asInt); err != nil {
-		intErr = err
-	} else {
-		*r = ResultType(asInt)
-		return nil
-	}
-
-	var asString string
-
-	if err := json.Unmarshal(data, &asString); err != nil {
-		return fmt.Errorf("unsupported value type, neither int nor string: %v", multierror.Append(intErr, err).ErrorOrNil())
-	}
-
-	switch asString {
-	case "TaskRunResult":
-		*r = TaskRunResultType
-	case "PipelineResourceResult":
-		*r = PipelineResourceResultType
```

Comment:
removal of this may change current users' PipelineResourceResult fall into unknowntype result, maybe we should let users be aware of this. Since PipelineResourceResult is going to be removed

Reply:
Thanks @Yongxuanzhang , I updated this in the release note also :)

Reply:
After digging through the code, I don't think anything is writing to termination messages for anything other than TaskRunResults and InternalTektonResults. There's nothing actionable for users here, so I think a release note would be more confusing than helpful.